\* authors contributed equally.

## Abstract {#abstract}

Metagenome-assembled genome (MAG) binning methods have emerged as a promising approach to recover individual genomes of microbes simultaneously within a community from metagenomic data without time-consuming and bias-inducing culture. 
The transmission of mobile genetic elements (MGEs), such as plasmids and genomic islands (GIs), is a public health priority due to their clinical, agricultural, and environmental implications. 
Variable copy numbers and sequence compositions of MGEs creates a challenge for studying MGEs using MAG binning methods, and current tools has not been well assessed for its ability to recover MGEs. 
To systematically investigate this, we simulated a metagenome comprising 30 GI-rich and plasmid-containing bacterial genomes. 
MAGs were then recovered using 12 workflows and evaluated for recovery of MGE-associated antimicrobial resistance (AMR)/virulence factor (VF) genes. 
We showed that only 38-44% of GIs and 1-29% of plasmid were recovered and correctly binned. 
Most strikingly, no plasmid-borne VF or AMR genes were recovered and, only 0-45% within GIs were identified. 
We conclude that short-read MAGs are largely ineffective for the analysis of mobile genes, including those of public-health importance like AMR and VF genes. 
We propose that microbiome researchers should instead primarily utilise unassembled short reads and/or long-read approaches to more accurately analyse metagenomic data. 