## Abstract {#abstract}

**final numbers to add**

Metagenomic methods, in which all the DNA in a sample is simultanously sequenced, is an increasingly popular method in the life sciences.
These approaches have advantages over genomic or phenotypic methods as they do not require time-intensive and bias-inducing culturing steps.
This means a much greater diversity can be profiled with minimal _a priori_ assumptions.
Due to this strength, metagenomics is emerging as a key tool in public health microbiology for the surveillance of virulence and antimicrobial resistance (AMR) genes.
A particular priority is identifying associations between these genes and mobile genetic elements such as plasmids and genomic islands (GIs) as they facilitae the evolution and transmission of these traits.
Unfortunately, metagenomic data, even when assembled, results in a mixed set of DNA contigs from multiple organisms rather than resolved individual genomes.
Recently, methods have been developed that attempt to group these fragments into bins likely to have been derived from the same underlying genome.
These bins are commonly known as metagenome-assembled genomes (MAGs). 
MAG based approaches have been used to great effect in revealing huge amounts of previously uncharacterised microbial diversity.
These methods typically achieve this by grouping sequences with similar sequence composition and relative abundance in the metagenome.
Unfortunately, plasmids are often found at different copy numbers than their host genome.
Both plasmids and genomic islands also often feature significantly different sequence composition than the rest of the genome.
Due to these features, we hypothesise these types of sequences will be highly under-represented in MAG based approaches.

To evaluate this we generated a simulated metagenomic dataset comprised of 30 genomes with up to 16.65% of chrosomomal DNA consisting of GIs and 65 associated plasmids.
MAGs were then recovered from this data using 12 different MAG pipelines.
The recovery and correct binning of mobile genetic elements was then evaluated for each pipeline.
Across all pipelines, 81.9-94.3% of chromosomes were recovered and binned. 
However, only 37.8-44.1% of GIs and 1.5-29.2% of plasmids were recovered and correctly binned at >50% coverage.
In terms of AMR and VF genes associated with MGEs, 0-45% of GI-associated AMR genes and 0-16% of GI-associated VF genes were correctly assigned.
More strikingly, 0% of plasmid-borne VF or AMR genes were recovered.

This work shows that regardless of the MAG recovery approach used, plasmid and GI dominated sequences will disproportionately be left unbinned or incorrectly binned.
From a public health perspective, this means MAG approaches are unsuited for analysis of mobile genes, especially vital groups such as AMR and VF genes.
This underlines the utility of read-based and long-read approaches to thorougly evaluate the resistome in metagenomic data.
