## Introductory Paragraph {#abstract}

Metagenomic methods are a key tool in the life sciences as they allow simultaneous characterisation of the DNA in a sample without time-consuming and bias-inducing culturing.
Metagenome-assembled genome (MAG) binning methods have emerged as a promising approach to recover individual resolved genomes from metagenomes.
However, MAG binning has not been well assessed for its ability to recover key sequences of clinical and biological significance such as mobile genetic elements (MGEs) like plasmids and genomic islands (GIs) and their associated antimicrobial resistance (AMR) and virulence factor (VF) genes.
The variable copy number and sequence composition of these MGEs relative to the host genome makes them potentially problematic for MAG methods. 
To investigate this, we simulated a low-complexity metagenome comprising 30 GI-rich (up to 16.5%) and plasmid associated (65) bacterial genomes.
MAGs were then recovered using 12 current MAG pipelines and evaluated for recovery of MGE-associated AMR/VF genes.
Here we show that while 81.9-94.3% of chromosomes could be correctly recovered and binned, the same was true for only 37.8-44.1% of GIs and 1.5-29.2% of plasmid sequences.
More strikingly, no plasmid-borne VF or AMR genes were recovered and only 0-45% of GI-associated AMR genes or 0-16% of GI-associated VF genes.
Therefore, regardless of approach, short-read metagenome MAGs are not well suited for the analysis of mobile genes, especially key groups of public-health importance like AMR and VF genes.
