\* authors contributed equally.

## Abstract {#abstract}

Metagenomic methods enable the simultaneous characterisation of microbial communities without time-consuming and bias-inducing culturing. 
Metagenome-assembled genome (MAG) binning methods aim to reassemble individual genomes from this data. 
However, the recovery of mobile genetic elements (MGEs), such as plasmids and genomic islands (GIs), by MAGs has not been well characterised. 
Given the association of antimicrobial resistance (AMR) genes and virulence factor (VF) genes with MGEs, studying their transmission is a public health priority. 
The variable copy number and sequence composition of MGEs makes them potentially problematic for MAG binning methods. 
To systematically investigate this, we simulated a low-complexity metagenome comprising 30 GI-rich and plasmid-containing bacterial genomes. 
MAGs were then recovered using 12 current prediction pipelines and evaluated. 
While 82-94% of chromosomes could be correctly recovered and binned, only 38-44% of GIs and 1-29% of plasmid sequences were found. 
Strikingly, no plasmid-borne VF or AMR genes were recovered, and only 0-45% of AMR or VF genes within GIs. 
We conclude that short-read MAGs are largely ineffective for the analysis of mobile genes, including those of public-health importance like AMR and VF genes. 
We propose that microbiome researchers should utilise unassembled short reads and/or long-read approaches to fully characterise metagenomic data. 
