## Abstract {#abstract}

Metagenomic methods, in which all the DNA in sample is simultaneously sequenced, is an increasingly popular method in the life sciences. 
They have a major advantage over genomic or phenotypic methods, as they do not require time-intensive and bias-inducing culturing steps. 
This means a much greater diversity can be achieved with minimal _a priori_ assumptions. 
Due to this strength, metagenomics is emerging as a key tool in public health microbiology for surveillance of virulence and antimicrobial resistance (AMR) genes. 
The most important sequences for surveillance purposes are those associated with mobile genetic elements such as plasmids and genomic islands (GIs). 
Unfortunately, metagenomic data, even when assembled, results in complex mixed set of DNA fragments rather than nicely resolved individual genomes. 
Recently, methods have been developed that attempt to group these DNA fragments into bins that likely to have been derived from the same underlying genome. 
These bins are referred to as metagenome-assembled genomes (MAGs). 
MAG based approaches have been used to great effect in revealing huge amounts of previously uncharacterised microbial diversity.
These methods perform this grouping using aspects of the sequence composition and the relative abundance of that sequence in the dataset. 
Unfortunately, plasmids are often represented at different copy numbers than the corresponding chromosomes. 
Additionally, both plasmids and genomic islands often feature significantly different sequence composition than the rest of the source genome as a whole. 
Due to this we hypothesise, these types of sequences will be highly under-represented in MAG based approaches.

To evaluate this we generated a simulated metagenomic dataset comprised of 30 genomes with up to 16.65% of chrosomomal DNA consisting of GIs and 65 associated plasmids. 
MAGs were then recovered from this data using 12 different MAG pipelines. 
The recovery and correct binning of mobile genetic elements was then evaluated for each pipeline. 
Across all pipelines, 81.9-94.3% of chromosomes were recovered and binned. However, only 37.8-44.1% of GIs and 1.5-29.2% of plasmids were recovered and correctly binned at >50% coverage. 
In terms of AMR and VF genes associated with MGEs, 0-45% of GI-associated AMR genes and 0-16% of GI-associated VF genes were correctly assigned. 
More strikingly, 0% of plasmid-borne VF or AMR genes were recovered.

This work shows that regardless of the MAG recovery approach used, plasmid and GI dominated sequences will disproportionately be left unbinned or incorrectly binned. 
From a public health perspective, this means MAG approaches are unsuited for analysis of mobile genes, especially vital groups such as AMR and VF genes. 
This underlines the utility of read-based and long-read approaches to thorougly evaluate the resistome in metagenomic data.