## Abstract {#abstract}

# Motivation

Metagenomic methods emerges as a key tool in public health microbiology for surveillance of virulence and antimicrobial resistance (AMR) genes. 
Unfortunately, metagenomic data, even when assembled, results in a complex, mixed set of DNA fragments rather than nicely resolved individual genomes. 
Recently, metagenome-assembled genomes (MAGs) emerged as approaches that attempt to group these DNA fragments into bins that are likely derived from the same underlying genome. 
The most important sequences for surveillance purposes are those associated with mobile genetic elements such as plasmids and genomic islands (GIs). 
we hypothesise that due to the different copy number and sequence composition of plasmids and GIs compared to their corresponding chromosomes, these types of sequences will be highly under-represented in MAG-based approaches.

# Results

To evaluate this, we generated a simulated metagenomic dataset comprised of 30 genomes with up to 16.65% of the chrosomomal DNA consisting of GIs and 65 associated plasmids. 
MAGs were then recovered from this data using 12 different MAG pipelines and evaluated for recovery accuracies. 
Across all pipelines, 81.9-94.3% of chromosomes were recovered and binned. However, only 37.8-44.1% of GIs and 1.5-29.2% of plasmids were recovered and correctly binned at >50% coverage. 
In terms of AMR and VF genes associated with MGEs, 0-45% of GI-associated AMR genes and 0-16% of GI-associated VF genes were correctly assigned. 
More strikingly, 0% of plasmid-borne VF or AMR genes were recovered.
This work shows that regardless of the MAG recovery approach used, plasmid and GI dominated sequences will disproportionately be left unbinned or incorrectly binned. 
From a public health perspective, this means MAG approaches are unsuited for analysis of mobile genes, especially vital groups such as AMR and VF genes. 
This underlines the utility of read-based and long-read approaches to thoroughly evaluate the resistome in metagenomic data.