## Abstract {#abstract}

Metagenomics, in which all the DNA in a sample is simultaneously sequenced, is a well established approach in the life sciences.
It offers major advantages over isolate-based genomic or phenotypic methods as it has minimal _a priori_ assumptions and doesn't require time-intensive and bias-inducing culturing. 
Due to this metagenomics is emerging as a key tool in public health microbiology for virulence factors (VF) and antimicrobial resistance (AMR) gene surveillance and rapid diagnostics.
In particular, these efforts are focused on genes associated with mobile genetic elements such as plasmids and genomic islands (GIs).


However, metagenomic sequencing and assembly results in a complex, mixed set of DNA fragments derived at random from all the genomes in the sample rather than resolved individual genomes preventing many useful analyses.
To address this, a range of methods have been developed that group these assembled DNA fragments, on the basis of shared sequence composition and abundance, into bins that are likely to have have derived from the same underlying genome.
These bins are typically referred to as metagenome-assembled genomes (MAGs).
MAG-binning methods have been used to great effect in revealing huge amounts of previously uncharacterised microbial diversity.


Unfortunately, mobile genetic elements are often present in different copy numbers, repetitively, and/or with significantly difference sequence composition relative to their source genome.
As MAG-binning approaches are based around these characteristics it is unclear how well they will perform for these mobile genetic elements of critical public health importance.


To evaluate this, we generated a simulated metagenomic dataset comprised of 30 genomes with up to 16.65% of the chrosomomal DNA consisting of GIs and 65 associated plasmids. 
MAGs were then recovered from this data using 12 different MAG-binning pipelines and the correct binning of mobile genetic elements evaluated.
Across all pipelines, 81.9-94.3% of chromosomal sequences were recovered and binned. 
However, only 37.8-44.1% of GIs and 1.5-29.2% of plasmids were recovered and correctly binned at >50% coverage. 
In terms of AMR and VF genes associated with MGEs, 0-45% of GI-associated AMR genes and 0-16% of GI-associated VF genes were correctly assigned. 
More strikingly, 0% of plasmid-borne VF or AMR genes were recovered.


This work shows that regardless of the MAG-binning approach used, plasmid and GI-dominated sequences will disproportionately be left unbinned or incorrectly binned. 
From a public health perspective, this means MAG approaches are unsuited for analysis of mobile genes, especially vital groups such as AMR and VF genes. 
This underlines the utility of read-based and long-read approaches to thoroughly evaluate the resistome in metagenomic data.
