## Introductory Paragraph {#abstract}

Metagenomic methods are a key, and increasingly popular, tool in the life sciences, as they enable simultaneous characterization of all microbes in a community without time-consuming and bias-inducing culturing. 
Metagenome-assembled genome (MAG) binning methods have emerged as a promising approach to recover individual resolved genomes from metagenomic data.
However, MAG binning has not been well assessed for its ability to recover key sequences of clinical/agricultural/environmental significance such as mobile genetic elements (MGEs), like plasmids and genomic islands (GIs). 
Certain antimicrobial resistance (AMR) genes and virulence factor (VF) genes are noted to be disproportionately associated with mobile sequences and studying their transmission has become a public health priority. 
However, the variable copy number and sequence composition of MGEs relative to the host genome makes them potentially problematic for MAG binning methods. 
To systematically investigate this, we simulated a low-complexity metagenome comprising 30 GI-rich and plasmid-associated bacterial genomes. 
MAGs were then recovered using 12 current MAG prediction pipelines and evaluated for recovery of MGE-associated AMR/VF genes. 
Here we show that while 82-94% of chromosomes could be correctly recovered and binned, only 38-44% of GIs were recovered and, even more notably, only 1-29% of plasmid sequences were found. 
Most strikingly, no plasmid-borne VF or AMR genes were recovered and within GIs, only between 0-45% of AMR or VF genes were identified. 
We therefore conclude that short-read MAGs are particularly not well suited for the analysis of mobile genes, including those of public-health importance like AMR and VF genes. 
We propose that microbiome researchers should instead utilise unassembled read-based and/or long-read approaches to more accurately analyze metagenomic data.
