## Abstract {#abstract}

Metagenomic methods, in which all the DNA in sample is simultaneously sequenced, is an increasingly popular method in the life sciences. 
They have a major advantage over genomic or phenotypic methods, as they do not require time-intensive and bias-inducing culturing steps. 
This means a much greater diversity can be achieved with minimal a priori assumptions. 
Due to this strength, metagenomics is emerging as a key tool in public health microbiology for surveillance of virulence and antimicrobial resistance (AMR) genes. 
The most important sequences for surveillance purposes are those associated with mobile genetic elements such as plasmids and genomic islands (GIs). 
Unfortunately, metagenomic data, even when assembled, results in complex mixed set of DNA fragments rather than nicely resolved individual genomes. 
Recently, methods have been developed that attempt to group these DNA fragments into bins that likely to have been derived from the same underlying genome. 
These bins are referred to as metagenome-assembled genomes (MAGs). 
MAG based approaches have been used to great effect in revealing huge amounts of previously uncharacterised microbial diversity.
These methods perform this grouping using aspects of the sequence composition and the relative abundance of that sequence in the dataset. 
Unfortunately, plasmids are often represented at different copy numbers than the corresponding chromosomes. 
Additionally, both plasmids and genomic islands often feature significantly different sequence composition than the rest of the source genome as a whole. 
Due to this we hypothesise, these types of sequences will be highly under-represented in MAG based approaches.

To evaluate this we generated a simulated metagenomic dataset comprised of genomes with large numbers of plasmids and considerable proportion of chromosomal DNA consisting of GIs at varying relative abundances. 
MAGs were then recovered from this synthetic data using a variety of different established MAG pipelines and parameterisations and correct binning of plasmid and GI sequences calculated relative to the genomes as a whole. 
We show that regardless of the MAG approach used, plasmid and GI dominated sequences will systematically be left unbinned (lost) or incorrectly binned. 
This indicates the importance of read based approaches for thorough evaluation of resistome complements in metagenomic data.
