\* authors contributed equally.

## Abstract {#abstract}

Metagenomic methods are an important tool in the life sciences, as they enable simultaneous characterisation of all microbes in a community without time-consuming and bias-inducing culturing. 
Metagenome-assembled genome (MAG) binning methods have emerged as a promising approach to recover individual genomes from metagenomic data.
However, MAG binning has not been well assessed for its ability to recover mobile genetic elements (MGEs), such as plasmids and genomic islands (GIs), that have very high clinical/agricultural/environmental importance. 
Certain antimicrobial resistance (AMR) genes and virulence factor (VF) genes are noted to be disproportionately associated with MGEs, making studying their transmission a public health priority. 
However, the variable copy number and sequence composition of MGEs relative to the majority of the host genome makes them potentially problematic for MAG binning methods. 
To systematically investigate this, we simulated a low-complexity metagenome comprising 30 GI-rich and plasmid-containing bacterial genomes. 
MAGs were then recovered using 12 current prediction pipelines and evaluated for recovery of MGE-associated AMR/VF genes. 
Here we show that while 82-94% of chromosomes could be correctly recovered and binned, only 38-44% of GIs were recovered and, even more notably, only 1-29% of plasmid sequences were found. 
Most strikingly, no plasmid-borne VF or AMR genes were recovered and within GIs, only between 0-45% of AMR or VF genes were identified. 
We conclude that short-read MAGs are largely ineffective for the analysis of mobile genes, including those of public-health importance like AMR and VF genes. 
We propose that microbiome researchers should instead utilise unassembled read-based and/or long-read approaches to more accurately analyse metagenomic data.
