## Introduction {#intro}

Metagenomics, the sequencing of fragments of DNA from within an environmental sample, is widely used for characterising viral and microbial communities [@doi:10.1073/pnas.202488399; @doi:10.1038/nbt.3935]. 
By randomly sampling from the total genomic content these methods allow researchers to simultaneously profile the functional potential and the taxonomic identity of a large proportion of the organisms in a sample. 
Metagenomic techniques are now bein used to profile antimicrobial resistance (AMR) and pathogen virulence. 
These approaches have been instrumental in developing our understanding of the distribution and evolutionary history of AMR genes [@doi:10.1016/j.cell.2014.08.032; @doi:10.1016/j.mib.2007.08.009; @doi:10.1038/nature10388], as well as tracking pathogen outbreaks [@doi:10.1001/jama.2013.3231].

While long-read DNA sequencing technology (e.g., Oxford Nanopore [@doi:10.1111/1755-0998.12324], PacBio [@doi:10.1126/science.1162986]) is now being used for metagenomic sequencing [@doi:10.1093/gigascience/giz043; @doi:10.1186/s12866-019-1500-0],  high-throughput sequencing of relatively short reads (150-250bp) in platforms such as the Illumina MiSeq currently dominate metagenomic analyses.
Inference of taxonomic and functional diversity can be assessed directly from sequenced reads using reference databases and BLAST-based sequence similarity search tools (e.g. DIAMOND [@doi:10.1038/nmeth.3176]), read mapping (e.g. Bowtie 2 [@doi:10.1038/nmeth.1923]), Hidden Markov Models (e.g. HMMER3 [@doi:10.1093/bioinformatics/btt403]) or k-mer hashing (e.g. CLARK [@doi:10.1186/s12864-015-1419-2]). 
These read-based approaches allow analysis of all reads with detectable similarity to genes of interest even if the organism has relatively low abundance in the sample.
Since these reads are shorter than most genes, however, read-based methods provide very little information about the genomic organisation of genes.
This lack of contextual information is particularly problematic in the study of AMR genes and virulence factors as the genomic context plays a role in function [@doi:10.1128/AAC.01710-09], selective pressures [@doi:10.1016/j.tim.2006.02.006], and the likelihood of lateral gene transfer (LGT) [@doi:10.1111/j.1574-6976.2011.00273.x].

Sequence assembly is often used to genereate information about genomic context [@doi:10.1038/nature02340]. 
de Bruijn graph-based assemblers have been developed to handle the particular challenges of this type of data including metaSPAdes [@doi:10.1101/gr.213959.116] , IDBA-UD [@doi:10.1093/bioinformatics/bts174], and megahit [@doi:10.1093/bioinformatics/btv033].
A crucial challenge in metagenomic analysis is that reads from different origanisms must be distentangled to avoid hybrid assemblies.
A common way to deal with this challenge is to assign all contigs from a given source genome to a cluster or "bin" based on similarities in the relative abudance and sequence composition.
These resulting bins are often known as metagenome-assembled genomes (MAGs). 
This binning is typically performed by grouping all the contigs with similar abundance and similar sequence composition into the same bin. 
A range of tools have been released to perform this binning including CONCOCT [@doi:10.1093/bioinformatics/btw290], MetaBAT 2 [@doi:10.7287/peerj.preprints.27522v1], and MaxBin 2 [@doi:10.1093/bioinformatics/btv638]. 
There is also the meta-binning tool DAS Tool [@doi:10.1038/s41564-018-0171-1] which combines predictions from multiple binning tools together. 
These MAG approaches have been used to great effect in unveiling huge amounts of previously uncharacterised genomic diversity [@doi:10.1038/nature14486; @doi:10.1038/s41564-017-0012-7; @doi:10.1101/489443].

An issue with MAGs is that there are loss of information at both the metagenomic assembly and binning steps.
This compounded data loss means that only a relatively small proportion of reads are successfully assembled and binned in large complex metagenome datasets, for example, 24.2-36.4% of reads from permafrost [@doi:10.1038/s41586-018-0338-1] and soil metagenomes [@doi:10.1038/s41564-019-0449-y]. 
Additionally, a large number of detected genomes are not reconstructed at all with ~23% of all detected genomes recovered in some examples [@doi:10.1038/s41564-019-0449-y]. 
The Critical Assessment of Metagenome Interpretation (CAMI) challenge's (https://data.cami-challenge.org/) Assessment of Metagenome BinnERs (AMBER) [@doi:10.1093/gigascience/giy069] assesses the global completeness and purity of recovered MAGs across methods.
However, to our best knowledge, there hasn't been a specific assessment of the impact of metagenomic assembly and binning on the loss of specific genomic elements.
In particular, the impact on mobile genetic elements (MGEs), such as genomic islands (GIs) and plasmids, which can be of great health and research importance, has not been evaluated.

Genomic islands (GIs) are clusters of genes that are known or predicted to have been acquired through LGT events. 
GIs can arise following the integation of MGEs, such as integrons, transposons, integrative and conjugative elements (ICEs) and prophages (integrated phages) [@doi:10.1038/nrmicro2350; @doi:10.1038/nrg3962]; these elements disproportionately encode virulence factors [@doi:10.1371/journal.pone.0008094] and are a major mechanism of LGT of AMR genes [@doi:10.3389/fmicb.2016.00173; @doi:10.1016/j.plasmid.2015.01.001]. 
GIs often have different nucleotide composition compared to the rest of the genome [@doi:10.1038/nrmicro2350].
This compositional difference is exploited by tools designed to detect GIs such as SIGI-HMM [@doi:10.1186/1471-2105-5-22] and IslandPath-DIMOB [@doi:10.1093/bioinformatics/bty095]. 
GIs may exist as multiple copies within a genome [@doi:10.1093/bib/bby042] leading to potential difficulties in correctly assembling these regions in metagenome assemblies as well as likely biases in the calculation of coverage statistics. 

Plasmids are circular or linear extrachromosomal self-replicating pieces of DNA. 
Similar to GIs, plasmids sequence composition are markley different compared to the genome they are associated with [@doi:10.1093/bioinformatics/btq299; @doi:10.1093/molbev/msp281]. 
This is largely attributable to their repetitive sequences, variable copy number, and different selection pressures [@doi:10.1038/s41559-016-0010; @doi:10.1128/AAC.00235-15].
Plasmids are of great importance, as they are a major source of the lateral dissemination of AMR genes throughout microbial ecosystems [@pmid:26603922; @doi:10.3389/fmicb.2016.00173]. 
Due to these reasons, the correct assembly of DNA sequence of plasmid origin has proven to be difficult from short read data [@doi:10.1099/mgen.0.000128,@doi:10.1099/mgen.0.000206].

GIs and plasmids pose significant challenges in MAG recovery due to their unusual sequence compositon and relative abundance; as these MGEs are key to the function and spread of pathogenic traits such as AMR and virulence, it is vital that we assess the impact of metagenome assembly and binning on the representation of these specific elements. 
This is particularly important with the increasing popularity of MAG approaches within public-health research. 
Therefore, to address this issue we performed an analysis of GI and plasmid recovery accuracy across a set of state-of-the-art short-read metagenome assembly and binning approaches using a simulated metagenome comprised of GI- and plasmid-rich taxa.
