## Results {#results}

The overall ability of MAG methods to recapitulate the original chromosomal source genome results varied widely.
We consider the identity of a given MAG bin to be that of the genome that composes the largest proportion of sequence within that bin.
In other words if a bin is identifiably 70% species A and 30% species B we consider that to be a bin of species A.
Ideally, we wish to generate a single bin for each source genome comprised of the entire genome and no contigs from other genomes.
Some genomes are cleanly and accurately binned regardless of the assembler and binning method used (Fig. (@fig:supspeciescov)).
Specifically, greater than 90% of Streptomyces parvulus (minimum 91.8%) and Clostridium baratii (minimum 96.4%) chromosomes are represented in individual bins across all methods.
However, no other genomes were consistently recovered by all methods for more than a third of the chromosomes.
The three Streptococcus genomes were particularly problematic with the best recovery for each ranging from 1.7% to 47.49%.

![Top Species Coverage](images/s1_species_top_coverage.png){#fig:supspeciescov}

In terms of assembler, megahit resulted in the highest median chromosomal coverage across all binners (81.9%) with metaSPAdes performing worst(76.8%) (Fig. (@fig:1topcoverage)).
In terms of binning tool, CONCOCT performed very poorly with a median 26% coverage for top hit per bin, followed by maxbin2 (83.1%), and metabat2 (88.5%).
It is perhaps unsurprising that the best performing binner in terms of bin top hit coverage was the metabinner DAS Tool that combines predictions from the other 3 binners (94.3% median top hit chromosome coverage per bin; (Fig. (@fig:1topcoverage))).

![Chromosomal coverages of most prevalent genome in each bin across binners and metagenome assemblies. Of the 3 assemblers (y-axis), megahit resulted in the highest median chromosomal coverage (x-axis) across all binners (colored bars) at 81.9% with metaSPAdes performing the worst (76.8%). Of the 4 binners, CONCOCT (blue) performed poorly with a median coverage, followed by maxbin2 (yellow), metabat2 (red) and DASTool (green) performing the best. Diamonds in the figure represents outliers and error bar represent 1st and 3rd quartiles.](images/1a_top_coverage.png){#fig:1topcoverage width="5in"}


Bin purity, i.e. the number of genomes present in a bin at >5% coverage, was largely equivalent across assemblers (Fig. (@fig:1purity)), with a very marginally higher purity for IDBA.
In terms of binning tool, however, maxbin2 proved an outlier with nearly twice as many bins containing multiple species as the next binner.
The remaining binning tools were largely equivalent, producing chimeric bins at approximately the same rates.

![Figure 2: Distribution of bin purities across assemblers and binners. The total number of genomes present in a bin at >5% coverage (y-axis) was largely equivalent across assemblers (x-axis). In term of binning tools, maxbin2 (orange) produced nearly twice as many bins containing multiple species compared to CONCOCT (blue), metabat2 (red) and DASTool (green), which all produced chimeric bins at roughly the same rate.](images/1b_purity.png){#fig:1purity width="5in"}

Regardless of method, a very small proportion of plasmids were correctly binned in the bin that mostly contained chromosomal contigs from the same source genome.
Specifically, between 1.5% (IDBA-UD assembly with DAS Tool bins) and 29.2% (metaSPAdes with CONCOCT bins) were correctly binned at over 50% coverage.
In terms of metagenome assembly, metaSPAdes was by far the most successful assembler at assembling plasmids with 66.2% of plasmids identifiable at greater than 50% coverage.
IDBA-UD performed worst with 17.1% of plasmids recovered, and megahit recovered 36.9%.
If the plasmid was successfully assembled, it was placed in a bin by maxbin2 and CONCOCT, although a much smaller fraction correctly binned (typically less than 1/3rd).
Interestingly, metabat2 and DAS Tool binners were a lot more conservative in assigning plasmid contigs to bins; however, of those assigned to bins nearly all were correctly binned ( Fig. (@fig:2plasmids))

![Plasmid Coverage](images/2_plasmid_coverage.png){#fig:2plasmids width="5in"}

GIs displayed a similar pattern of assembly and correct binning performance as plasmids (Fig. (@fig:3gis)).
These sequences were assembled uniformly badly (37.8-44.1%) with metaSPAdes outperforming the other two assembly approaches.
For CONCOCT and maxbin2 binning tools all GIs that were assembled were assigned to a bin although the proportion of binned GIs that were correctly binned was lower than for DAS Tool and metabat2.
DAS Tool, metabat2 and CONCOCT did not display the same precipitous drop between those assembled and those correctly binned as was observed for plasmids.
In terms of overall correct binning with the chromosomes from the same genome the metaSPAdes assembly with CONCOCT (44.1%) and maxbin2 (43.3%) binners performed best.

![Genomic Island Coverage. Genomic islands (GIs) showed a similar poor recovery as plasmids. Generally, <40% were correctly binned to the same bin that contained chromosomal contigs from the same source genome regardless of binning (x-axis) and assembly (facet) methods to >50% coverage (y-axis). MetaSPAdes performed the best at assembling GIs (blue). Maxbin2 and CONCOCT placed GIs in a bin majority of the time (orange) however a very small fraction was correctly binned (green). Generally, GIs were correctly binned better than plasmids with DAS Tool, metabat2 and CONCOCT.](images/3_gi_coverage.png){#fig:3gis width="5in"}

In term of gene content, we first explored the ability to find open reading frames (ORFs) within MAGs.
Overall, the total number of predicted ORFs in MAGs followed a similar trend as the chromosomal coverage and purity (Fig. (@fig:4geneContent)).
Of the four binning tools, CONCOCT performed the worst, finding <30% of the number of ORFs in our reference genomes.
Metabat2 performed second worst at ~80%.
DAS Tool recovered a similar number to our reference and Maxbin2 seemed to predicted 7-46% more genes.
The Assembler method did not significantly impact the number of genes predicted with the exception of Maxbin2, in which IDBA_UD was the closest to reference and metaSPAdes predicted 46% more ORFs.

![Predicted Gene Content. The total number of open reading frames (ORF) predicted followed the same trend as chromosomal coverage and purity. The assemblers (colored bars) did not contribute to a big variance in the number of ORFs. Of the 4 binners, CONCOCT recovered <30% of our reference genome ORFs. DAS Tool and metabat2 predicted a similar number as our reference genomes. ](images/4geneContent.png){#fig:4geneContent width="8in"}

First, we focused on the ability of MAGs to recover clinically relevant AMR genes (Fig. (@fig:5AMRGenePercentRecoveryStage)).
After the assembly stage, we were only able to recover between ~49-55% of the AMR genes predicted in our reference genomes regardless of the assembly tool used, with metaSPAdes performing marginally better.
Binning the contigs resulted in a ~1-15% loss in AMR gene recovery with concoct-metaSPAdes pair performing the best at 1% loss and dasTool-megahit performing the worst at 15% reduction of AMR genes recovered.
Only 24% - 40% of all AMR genes were correctly binned.
Maxbin2 and IDBA_UD performing the worst (24%) and concoct-metaSPAdes performing the best (40%).

![Percent of reference antimicrobial resistance genes (AMR) recovered across assemblers and binners. The proportion of reference AMR genes recovered (y-axis) was largely similar across assembly tools (blue), at roughly 50% with metaSPAdes performing marginally better. Binning tools resulted in a small reduction in AMR genes recovered (orange), however only 24-40% of all AMR genes were correctly binned (green). MetaSPAdes-CONCOCT was the best performing pair. ](images/5AMRGenePercentRecoveryStage.png){#fig:5AMRGenePercentRecoveryStage width="15in"}

Moreover, focusing on only the genes that are correctly binned with respect to the reference replicon assigned to that bin (Fig. (@fig:6AMRGenePercentRecoveryCorrectlyBinned)).
MAGs was able to correctly bin only 30%-53% of all chromosomally located AMR genes (n=120), 0-45% of genomic island located AMR genes (n=11) and none of the plasmid located AMR genes (n=20).
Majority of the remaining genes was unbinned and some were incorrectly binned.

![Percent of correctly binned AMR genes recovered in each genomic region. Metagenome assembled genomes (MAGs) were best at recovering chromosomally located AMR genes (light blue) regardless of assembler-binner pairs. GI recovered showed a bigger variation in %recovery between tools (light green). None of the assembler-binner pair was able to recover plasmid located AMR genes (orange). Majority of the remaining genes was unbinned and some were incorrectly binned.](images/6AMRGenePercentRecoveryCorrectlyBinned.png){#fig:6AMRGenePercentRecoveryCorrectlyBinned width="15in"}

Aside from AMR genes, we also examined virulence factors (VF) in our dataset using the Virulence Factor Database (VFDB).
We saw a similar trend as AMR genes (Fig. (@fig:7VFGenePercentRecoveryStage)).
There was minor differences for VF recovered after the assembly stage, with 56-64% of VF recovered.
Megahit was able to produce marginally better recovery with VFs.
Similarly to AMR genes, binning the contigs again reduced the recovery between 4-26%, with dasTool-megahit performing the worst (26%) and concoct-metaSPAdes performing the best (4%).
Unlike AMR genes, majority of the binned VF genes were correctly assigned to the right bin.
Concoct-metaSPAdes again performed best 43% of all VFs correctly assigned.

![Percent of reference virulence factor (VF) genes recovered across assemblers and binners. The proportion of reference VF genes recovered (y-axis) exhibited a similar trend as AMR genes. Recovery was greatest after the assembling stage (blue), with megahit performing best. Binning tools resulted in a larger reduction in VF genes recovered (orange) compared to AMR genes. However, in majority of cases, VF genes that are binned are correctly binned (green). MetaSPAdes-CONCOCT was again the best performing pair.](images/7VFGenePercentRecoveryStage.png){#fig:7VFGenePercentRecoveryStage width="15in"}

Again looking at the location of the VFs that were correctly assigned (Fig. (@fig:8VFGenePercentRecoveryStage)).
MAGs was able to correctly bin majority (73%-98%) of all chromosomally located VF genes (n=757), 0-16% of genomic island located VF genes (n=809) and again none of the plasmid located VF genes (n=3).

![Percent of correctly binned VF genes Recovered in each genomic region. Metagenome assembled genomes (MAGs) were again best at recovering chromosomally located VF genes (light blue), able to correctly bin majority of chromosomally located VFs. GIs recovered again performed very poorly (light green) and again none of the plasmid located AMR genes (orange) was correctly binned.](images/8VFGenePercentRecoveryStage.png){#fig:8VFGenePercentRecoveryStage width="15in"}
