## Conclusions {#conclusions}

Using a simulated metagenome, this study has shown that MAG-based approaches provide a useful tool to study a bacterial speciesâ€™ core chromosomal elements, but have severe limitations in the recovery of MGEs.
The majority of these MGEs will either fail to be assembled or be incorrectly binned.
The consequence of this is the disproportionate loss of key public-health MGE-associated VFs and AMR genes.
As these genes have a much higher propensity for lateral gene transfer between unrelated bacteria it is critical to highlight that MAG approaches alone are insufficient to thoroughly profile them.
Within public-health metagenomic research it is vital we utilize MAGs in conjunction with other methods (e.g. targeted AMR [@doi:10.1099/mgen.0.000131], plasmid specialised assembly approaches [@doi:10.1101/2020.01.12.903252], and read-based sequence homology search [@doi:10.1038/nmeth.3176]) before drawing biological or epidemiological conclusions.
