## Discussion {#discussion}

In this paper, we evaluated the ability and accuracy of metagenome-assembled genomes (MAGs) to correctly recover mobile genetic elements (i.e. genomic islands and plasmids) from metagenomic samples across different tools used to assemble and bin MAGs.

Overall, the best assembler-binner pair was megahit-DASTOOL in term of both chromosomal coverage (94.3%) and bin purity (1).
Looking at genomes with the lowest coverage, the three Streptococcus genomes that was recovered poorly are likely due to their similarity.
This supports the intuition that MAG recovery approaches struggle to distinguish closely related species.
While CONCOCT performed significantly worse than other binners in terms of chromosomal coverage and bin purity, we did notice that CONCOCT seems to display a trend of generating many small partial bins.
Potentially, CONCOCT binning could be used to distinguish closely related species but at a cost of more fragmented genomes.
 
While the overall recovery and binning of chromosomes were acceptable, we were specifically interested in the ability of MAG methods to appropriately recover MGEs. 
This was due to the importance of MGEs in the function and spread of pathogen traits such as AMR and virulence, as well our hypothesis that their sequence characteristics (composition and copy number) would prove difficult to bin.
Unfortunately, our hypothesis was confirmed, despite the metagenomic assembly approach or MAG binning method used both plasmids and GIs were disproportionately lost compared to chromosomes in general.
At best (via metaSPAdes and CONCOCT) 29.2% of plasmids and 44.1% GIs were identifiable at >50% coverage in the correct bin (i.e. grouped with a bin that was mostly made up of contigs from the same genome).
The >50% coverage requirement might have been a high-bar and there is a possibility that more GIs and plasmids were recovered but in very incomplete forms.

This poor result is not unexpected as genomic islands and plasmids have known divergent compositional features and are often repetitive with variable copy numbers relative to the chromosomes.
Furthermore, the difference between the percentages suggest that binning plasmids are harder than binning GIs.
This difference is partially attributal to the known difficulties in assembly of plasmids from short-read data [@pmid:29177087].
Therefore, binning efficiency might improve if we use DNA sequencing and assembly methods optimised for recovering plasmids [@doi:10.1099/mgen.0.000128].

Due to the importance of mobile genetic elements to disseminate clinically relevant AMR genes and VFs, we explored whether or not MAG approaches can be used to provide useful insight LGT of these genes.
With respect to AMR genes, MAG methods were able to recover roughly 40% of all AMR genes present in our reference genomes.
We noted a sharp drop in the number of AMR genes detected between assemblies and MAGs, suggesting that many of these genes were left in the unbinned portion.
Overall, CONCOCT-metaSPAdes combination, while it did not recover the highest amount of AMR genes at the assembly stage, performed the best in correctly binning an AMR gene to the right species.
Regardless of tools, chromosomally-located AMR genes were most frequently correctly binned (as expected from the relative performance of MAGs at recovering chromosomes).
While there was a lot of variability in performance, AMR genes located on GIs were correctly binned slightly less well than chromosomally located AMR genes.
This variability might be explained by the fact there were only 11 GI located AMR genes in our reference genomes.
All 20 of the plasmid-borne AMR genes were assembled but none were placed into MAG bins.
This included high-threat MGEs-based AMR genes such as the KPC and OXA carbapenemases.

Virulence factors had shown a similar trend as AMR genes, recovering ~63% of virulence factors present in the reference genome.
There still is a sharp decline in the number of VF detected between assemblies and MAGs and CONCOCT-metaSPAdes again produced the highest binning accuracy.
MAGs were also able to correctly bin majority (73%-98%) of chromosomally located VF genes to the right species.
However, MAGs performed much worse in correctly recovering GI located and plasmid located VFs, with <16% of GI VFs (n=809) correctly recovered and none of the plasmid VFs (n=3).
This drastic reduction in recovery accuracy of mobile elements, especially GI, is expected.
Previous studies has found that VFs are disproportionally present on GIs[@doi:10.1371/journal.pone.0008094], which might be the reason to why the recovery accuracy was worse compared to AMR genes.
Together, this and the AMR gene results suggests that MAG-based methods might be of limited utility in public health research focused on the transmission and dissemination of AMR genes and VFs. 

It should also be noted that while CONCOCT performed the best in terms of recovery of both chromosomes and MGEs, it created lots of relatively clean but fragmentary partial MAGs.
While this might be ideal for some users, caution should be taken in using CONCOCT when assuming a bin represents a whole genome.

With the recovery of plasmids, GIs, VFs, and AMR genes the same pattern was observed, a progressive loss of data in each analytical step.
The act of metagenomic assembly itself generally resulted in the loss of the majority of these elements/genes regardless of assembly method used.
Across binning tools, the binning process resulted in further loss with a large proportion of MGEs and genes left unbinned.
Finally, only a very small proportion of these elements/genes were generally correctly binned with the appropriate host chromosomes.
While the concept that analysis is "lossy" and that the more analysis you do the more of the input data you are likely to lose is fairly well known it is rarely explicitly stated.
Indeed, this is one of the reasons why the huge amount of redundancy in metagenomic sequencing is necessary i.e. many more base-pairs of DNA than are in the underlying sample.
