## Discussion {#discussion}

In this paper, we evaluated the ability and accuracy of metagenome-assembled genome (MAGs) binning methods to correctly recover mobile genetic elements (i.e. genomic islands and plasmids) from metagenomic samples across different tools used to assemble and bin MAGs.
Overall, chromosomal sequences were binned well (up to 94.3% coverage and perfect bin purity by megahit-DASTool) with closely related genomes consistently badly binned across tools (e.g. Streptococcus species in Fig. @fig:coverphylo, @fig:purityphylo).
However, given the increased used of MAG methods and the importance of MGEs in the function and spread of pathogen traits such as AMR and virulence it is particulraly striking that regardless of metagenomic assembly approach or MAG binning method used, both plasmids and GIs were disproportionately lost compared to chromosomes in general. 
At best (with metaSPAdes and CONCOCT) 29.2% of plasmids and 44.1% of GIs were identifiable at >50% coverage in the correct bin (i.e. grouped with a bin that was mostly made up of contigs from the same genome). 
This poor result is congruent with the intuition that the divergent compositional features and repetitive nature of these MGEs is problematic for MAG methods.
The particularly poor plasmid binning performance is likely attributable to the known difficulties in assembly of plasmids from short-read data [@pmid:29177087].
Therefore, binning efficiency might improve if we use DNA sequencing and assembly methods optimised for recovering plasmids [@doi:10.1099/mgen.0.000128] (such as SCAPP [@doi:10.1101/2020.01.12.903252]).
While some MGEs were likely recovered in more partial forms (<50% coverage) use of these by researchers interested in selective pressures and lateral gene transfer could lead to inaccurate inferences. 

Given the growing use of MAG methods in infectious disease research and the importance of MGEs in the dissemination of clinically important AMR genes and VFs, we also specifically evaluated the binning of these gene classes.
MAG methods were able to recover, at best, 40% of all AMR genes and ~63% of VF genes (CONCOCT-metaSPAdes) present in the reference genomes.
While a majority of chromosomally located VF genes (73-98%) and AMR genes (%) were binned correctly only 16% of GI VFs (n=809), 0% of plasmid VFs (n=3), % of GI AMR genes (n=11), and 0% of plasmid AMR genes.
This included critical high-thread MGE-associated AMR genes such as the KPC and OXA carbapenemases. 
One potential caveat is that some AMR genes and VFs successfully assembled in the MAGs may no longer be annotated as such due to issues with ORF prediction (see suppl. discussion & Fig. @fig:geneContent). 
Previous studies have observed that ORF predictions in draft genomes are more fragmented, which can lead to downstream over- or under-annotation with functional labels depending on the approach used [@doi:10.1186/1471-2164-13-14]. 
Similarly, if the ORFs predicted in the MAGs differ in sequence or degree of fragmentation from the corresponding ORFs predicted in the original reference genomes (or are no longer predicted at all), this could impact recovery of AMR/VF predictions, even though the sequences themselves may be partially or fully present in the assembly.

There are 3 main limitations to this study: the drawbacks of metagenome simulation, the use of a single metagenome, and not evaluating whether using multiple replicates of a metagenome sample can help with recovery of these elements.
Metagenome simulation is unlikely to perfectly emulate all the noise and biases present in real metagenomic sequencing results. 
However, without a clear known ground truth and in the absence of an MGE/AMR focused mock metagenome simulation represents the most powerful technique for evaluation of bioinformatic analyses. 
This is why it has been widely adopted in thorough benchmarking experiments such as the CAMI challenges [@doi:10.1093/gigascience/giy069;@doi:10.1186/s40168-019-0633-6].
While multiple metagenomic samples may improve binning of MGEs [@doi:10.1126/sciadv.aax5727] most typical public-health and research MAG studies don't currently utilise this kind of approach.
Finally, a single simulated metagenome was used to keep the evaluation and the issues with accurate binning of MGEs as clear as possible.
As this simulated metagenome was relatively low-complexity (30 species) this was considered as a "best-case" evaluation, i.e., failure to recover elements for this dataset strongly indicates likely failure to recover them in any more complex metagenome.
