## Discussion {#discussion}

In this paper, we evaluated the ability and accuracy of metagenome-assembled genomes (MAGs) to correctly recover mobile genetic elements (i.e. genomic islands and plasmids) from metagenomic samples across different tools used to assemble and bin MAGs. 

Overall, the best assembler-binner pair was megahit-DASTOOL in term of both chromosomal coverage (94.3%) and bin purity (1). 
Looking at genomes with the lowest coverage, the 3 Streptococcus genomes were particularly problematic, likely due to their similarity, with the best recovery for each ranging from 1.7% to 47.49%. 
This suggest that MAGs might not be able to distinguish between closely related species. 
While CONCOCT performed significantly worse compared to the other binners in chromosomal coverage and bin purity, we did notice that CONCOCT seems to display a trend of generating lots of small partial bins. 
Perhaps CONCOCT bins might be able to distinguish between closely related species to a higher resolution (COMMENT: Small partial bins… what does it mean overall. is this assumption correct? Would it be able to distinguish closely related species?)

While the overall recovery of chromosomes were okay, we were interested in MAG’s ability to correctly bin mobile genetic elements due to their importance in the functions and spread of pathogenic traits such as AMR and virulence.
In term of plasmids, a very small proportion of plasmids were correctly binned regardless of the method (<33% at best). 
Similarly, the same trend exists for genomic islands (<43.3%). 
This poor result is not unexpected as genomic islands and plasmids have divergent composition features relative to the chromosomes. 
Furthermore, the difference between the percentages suggest that binning plasmids are harder than GIs. This difference might be due to the problem of plasmid assembly. 
Therefore, the binning efficiency might improve if we use an assembler targeted at assembling plasmids [@doi:10.1099/mgen.0.000128].

Due to the importance of mobile genetic elements to disseminate clinically relevant antimicrobial resistance genes and virulence factors, we explored whether or not MAGs can be used to provide useful lateral gene transfer insights. 

With respect to AMR genes, MAGs were able to recover roughly 40% of all AMR genes present in our reference genome. 
We noted a sharp drop in the number of AMR genes detected between assemblies and MAGs, suggesting that the these genes were left in the unbinned portion. 
Overall, CONCOCT-metaSPAdes combination, while did not recover the highest amount of AMR genes at the assembly stage, performed the best in correctly binning an AMR gene to the right species. 
It should be noted that CONCOCT seems to generated a lot of small partial bins. 
This might led to a better seperation between closely related sequence compositions resulting in the improved recovery we saw. 
Regardless of tools, chromosomally located AMR genes were recovered best and were able to be correctly binned. 
The accuracy of these were as expected given the accuracy of MAGs to recover chromosomes as discussed previously. 
With respect to mobile elements, the ability of MAGs to recovery genomic island located AMR genes varied across tools but the recovery accuracy is slightly worse than chromosomally localed AMR genes. 
However it should be noted that there were only 11 GI located AMR genes in our reference genome. 
While MAGs were able to detect all 20 plasmid-born AMR genes, none of these were placed in any of the bins. 
We specifically included a few high threat mobile element located AMR genes in our dataset: namely KPC and OXA carbapenemases that are of increasing prevalence in the clinics capable of rendering our last resort antibiotics useless. 
These genes were successfully detected from the metagenomics assembly, but they were not assigned to any bin. 
This could mean a limited ability for MAGs to be used in the public health research to pinpoint the lateral transfer of AMR genes and to conduct epidemiological analysis (COMMENT: does this make sense?). 

Virulence factors had shown a similar trend as AMR genes, recovering ~63% of virulence factors present in the reference genome. 
There still is an sharp decline in the number of VF detected between assemblies and MAGs and CONCOCT-metaSPAdes again produced the highest binning accuracy. 
MAGs were also able to correctly bin majority (73%-98%) of chromosomally located VF genes to the right species. 
However, MAGs performed much worse in correctly recovering GI located and plasmid located VFs, with <16% of GI VFs (n=809) correctly recovered and none of the plasmid VFs (n=3). 
This drastic reduction in recovery accuracy of mobile elements, especially GI, isn't unexpected. 
Previous studies has found that VFs are disproportionally present on GIs[@doi:10.1371/journal.pone.0008094], which might be the reason to why the recovery accuracy was worse compared to AMR genes. 

Looking at the total amount of predicted gene content, our best assembler-binner pair produced a similar number of predicted ORFs as our reference genomes. 
Interestingly, we still missed upwards of around half of AMR genes VFs. Perhaps theses predicted ORFs are fragmented, resulting in an ability to detect these genes in MAGs.

Lastly, previous works have shown that AMR genes that are on mobile genetic elements disproportionally encode secrete proteins. Given that the recovery of plasmid-borne genes were not great, we asked if MAGs would affect the ability to predict the subcellular localization of proteins. We found that the proportion of predicted localizations were very similar between MAGs and our reference genomes, suggesting that there is not a significant penalty to use MAGs as input for protein localization predictions. 
