## Conclusions {#conclusions}

Using a simulated medium-complexity metagenome, this study has shown that MAG-based approaches provides a useful tool to study a bacterial speciesâ€™ chromosomal elements, but have severe limitations in the recovery of MGEs.
The majority of these MGEs will either both fail to assemble or be incorrectly binned.
The consequence of this is the disproportionate loss of key public health priority genes like VF and AMR genes.
This is particularly acute as the VF and AMR genes found on these poorly recovered MGEs are generally considered the most important due to their propensity for lateral gene transfer between unrelated bacteria.
Therefore, it is vital that we utilize a combination of MAGs and other methods (e.g. read-based methods) in public health metagenomic research when short-read sequencing is used.
For example, targeted AMR [@doi:10.1099/mgen.0.000131], plasmid specialised assembly approaches [@doi:10.1101/2020.01.12.903252], and read-based sequence homology search [@doi:10.1038/nmeth.3176].
Without this, MAG-based methods are insufficient to thoroughly profile the resistome and provide vital epidemiological data for metagenomic data.
